### PR TITLE
[SMALLFIX] Generating separate client and server assembly jars.

### PR DESCRIPTION
**Pull request by jsimsa**

### CODE REVIEW

File: assembly/client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>alluxio-assembly</artifactId>
+    <groupId>org.alluxio</groupId>
+    <version>1.5.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>alluxio-assembly-client</artifactId>
+
```
Comment (calvinjia) [inline assembly/client/pom.xml]: Add `packaging`, `name`, and `description`? (same for server pom)

File: assembly/server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-underfs-gcs</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-underfs-glusterfs</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```
Comment (calvinjia) [inline assembly/server/pom.xml]: I think this will conflict w/ #5425 @apc999

File: assembly/client/pom.xml (hunk 1 above)
Comment (jsimsa) [inline assembly/client/pom.xml]: done

File: bin/alluxio [hunk 3]
```diff
@@ -259,47 +260,56 @@ function main {
...
   ;;
   "readJournal")
     CLASS="alluxio.master.journal.JournalTool"
+    CLASSPATH=${ALLUXIO_SERVER_CLASSPATH}
     ALLUXIO_SHELL_JAVA_OPTS+=" -Dalluxio.logger.type=Console"
     runJavaClass "$@"
   ;;
   "upgradeJournal")
     CLASS="alluxio.master.journal.JournalUpgrader"
+    CLASSPATH=${ALLUXIO_SERVER_CLASSPATH}
```
Comment (apc999) [inline bin/alluxio]: will this be overwritten by line 263? @jsimsa

**Pull request merged**